src/python-common/ceph/rgw/rgwam_core.py

```
@@ -386,101 +430,125 @@ def zone_op(self):
+        if realm_name in self.realm_op().list():
+            raise RGWAMException(f'Realm {realm_name} already exists')
+        if zonegroup_name in self.zonegroup_op().list():
+            raise RGWAMException(f'ZonegroupOp {zonegroup_name} already exists')
+        if zone_name in self.zone_op().list():
+            raise RGWAMException(f'Zone {zone_name} already exists')
+
+        # Create RGW entities and update the period
+        realm = self.create_realm(realm_name)
+        zonegroup = self.create_zonegroup(realm, zonegroup_name, zonegroup_is_master=True)
+        zone = self.create_zone(realm, zonegroup, zone_name, zone_is_master=True)
+        self.update_period(realm, zonegroup)
+
+        # Create system user, normal user and update the master zone
+        sys_user = self.create_system_user(realm, zonegroup, zone)
+        access_key = sys_user.keys[0].access_key
+        secret = sys_user.keys[0].secret_key
+        self.zone_op().modify(zone, zonegroup, None, None, access_key, secret)
+        self.update_period(realm, zonegroup)
+        self.create_normal_user(zonegroup, zone)
+
+        if start_radosgw:
+            # Instruct the orchestrator to start RGW daemons, asynchronically, this will
+            # call back the rgw module to update the master zone with the corresponding endpoints
+            realm_token = RealmToken(realm_name,
+                                     realm.id,
+                                     True,   # primary cluster
+                                     None,   # no endpoint
+                                     access_key, secret)
+            realm_token_b = realm_token.to_json().encode('utf-8')
+            realm_token_s = base64.b64encode(realm_token_b).decode('utf-8')
+            rgw_spec.rgw_realm_token = realm_token_s
+            rgw_spec.update_endpoints = True
+            self.env.mgr.apply_rgw(rgw_spec)
 
     def realm_new_zone_creds(self, realm_name, endpoints, sys_uid):
```

why is a system user getting created here? If doing a `radosgw-admin user info --uid ${sys_uid}` doesn't give you back the keys for sys_uid you should throw an error.

---

I have no idea in this case since this code is legacy and I didn't change this function.